spec/requests/v0/map_services_controller_request_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'map/security_token/service'
+
+RSpec.describe V0::MapServicesController, type: :request do
+  include ServiceAccountAuthorizationHelper
+
+  describe 'POST /v0/map_services/token' do
+    subject(:call_endpoint) { post v0_map_services_token_path, params:, headers: authorization_header }
+
+    let(:params) { { application: 'foo', icn: '42' } }
+    let(:authorization_header) do
+      service_account_authorization_header(scope: 'http://www.example.com/v0/map_services/token')
+    end
+
+    let(:sts_client) { instance_spy(MAP::SecurityToken::Service) }
```

Comment:
❓ I tend to reach for spies over going through the hassle of VCR cassettes and whatnot, however, I can see the benefits of both approaches. Does this team have a strong preference either way? I figured using a spy is acceptable given the `MAP::SecurityToken::Service` client that actually interacts with the STS service is already pretty thoroughly tested with VCR/webmock, so it seemed redundant to do so here.

Reply:
For controller/request specs, I tend towards mocking as little as possible. The reason is I conceptually think request specs should be as close to integration tests as possible. So for this spec, I would tend towards VCR. On the other hand if this were a model/service/some other unit spec for a class that leveraged `MAP::SecurityToken::Service`, then I would def say 'feel free to mock it' if it was sufficiently tested in the underlying class spec